main.go

```diff
@@ -182,47 +172,48 @@ func (s *MagicLinksService) authenticateHandler(w http.ResponseWriter, r *http.R
 		// a new Organization.
 		s.saveSession(w, r, intermediateSessionKey, resp.IntermediateSessionToken)
 
-		discoveredOrgs := make([]DiscoveredOrganization, len(resp.DiscoveredOrganizations))
-		for i := range resp.DiscoveredOrganizations {
-			discoveredOrgs[i] = DiscoveredOrganization{
-				OrganizationId:   resp.DiscoveredOrganizations[i].Organization.OrganizationID,
-				OrganizationName: resp.DiscoveredOrganizations[i].Organization.OrganizationName,
-			}
-		}
-		// Sort Organizations alphabetically.
-		slices.SortFunc(discoveredOrgs, func(a, b DiscoveredOrganization) int {
-			return cmp.Compare(a.OrganizationName, b.OrganizationName)
-		})
-
-		log.Println("Successfully authenticated with discovery")
-		if err := RenderTemplate(w, DiscoveredOrganizations, &TemplateData{
-			Email:                   resp.EmailAddress,
-			IsLogin:                 true,
-			DiscoveredOrganizations: discoveredOrgs,
-		}); err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
-		return
-	}
-
-	if tokenType == "multi_tenant_magic_links" {
-		resp, err := s.client.MagicLinks.Authenticate(ctx, &magiclinks.AuthenticateParams{
-			MagicLinksToken: token,
+		authenticatedEmail = resp.EmailAddress
+		responseDiscoveredOrgs = resp.DiscoveredOrganizations
+	} else if tokenType == "discovery_oauth" {
+		resp, err := s.client.OAuth.Discovery.Authenticate(ctx, &oauthdiscovery.AuthenticateParams{
+			DiscoveryOAuthToken: token,
 		})
 		if err != nil {
 			log.Printf("Error authenticating: %v\n", err)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		s.saveSession(w, r, sessionKey, resp.SessionToken)
-		log.Println("Successfully authenticated with multi_tenant_magic_links")
-		s.indexHandler(w, r)
+		s.saveSession(w, r, intermediateSessionKey, resp.IntermediateSessionToken)
+
+		authenticatedEmail = resp.EmailAddress
+		responseDiscoveredOrgs = resp.DiscoveredOrganizations
+	} else {
+		log.Printf("Error: unrecognized token type %s\n", tokenType)
```

Comment:
Curious in terms of consistency what the different usages of log.Printf vs. fmt.Printf are? since log.Printf was replaced above on line 165

Reply:
I had to look it up too, I wasn't super familiar, here the StackOverflow post: https://stackoverflow.com/questions/19646889/why-should-i-use-log-println-instead-of-fmt-println. Using `log.` apparently helps with concurrency (probably not a big problem for an example app). The other thing it does is add timestamps to the server, which is nice for debugging, like this:
![Screenshot 2024-10-22 at 10 33 30 AM](https://github.com/user-attachments/assets/b1759817-ad37-406a-94d1-da8c03e576f8)
Which means for an error especially, we should prefer to use `log`.